developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,247 @@
+.. _ipc4_extended_module_config:
+
+Ipc4_extended_module_config
+###########################
+
+The ipc4 extended module config is a place to store build-time known module metadata
+such as the module name, uuid or capabilities. This config is part of the output
+binary is located in extended manifest structure in the binary file and it is used
+by driver when setting up ipc message to dsp.
+
+Rimage builds extended module config based on module setting in config/xxx.toml.
+The following setting is for gain module.
+
+.. code-block:: bash
+
+        [[module.entry]]
+        name = "GAIN"
+        uuid = "61BCA9A8-18D0-4A18-8E7B-2639219804B7"
+        affinity_mask = "0x1"
+        instance_count = "40"
+        domain_types = "0"
+        load_type = "0"
+        module_type = "5"
+        auto_start = "0"
+        sched_caps = [1, 0x00008000]
+
+        # pin = [dir, type, sample rate, size, container, channel-cfg]
+        pin = [0, 0, 0xfeef, 0xf, 0xf, 0x45ff,
+                        1, 0, 0xfeef, 0xf, 0xf, 0x1ff]
+
+        # mod_cfg [PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS]
+        mod_cfg = [0, 0, 0, 0, 416, 914000, 48, 64, 0, 0, 0,
+                                1, 0, 0, 0, 416, 1321600, 192, 256, 0, 0, 0,
+                                2, 0, 0, 0, 416, 1786000, 192, 256, 0, 0, 0,
+                                3, 0, 0, 0, 416, 2333000, 48, 64, 0, 0, 0,
+                                4, 0, 0, 0, 416, 2910000, 192, 256, 0, 0, 0,
+                                5, 0, 0, 0, 416, 3441000, 192, 256, 0, 0, 0,
+                                6, 0, 0, 0, 416, 4265000, 192, 256, 0, 0, 0]
+
+Module entry definition
+***********************
+| **name**: name of the module
+| **uuid**: uuid of the module
+| **affinity_mask**: usable dsp core mask, bit N is for core N, 0x3 means core 0 and 1 can be used
+| **instance_count**: max module instance count can be active in fw
+| **domain_types**: two schedule domains: domain ll (0), and dp (1)
```

Comment:
I'd assume these are indexes of bits in the mask since you may define a module that works in both domains? But you provided value of "0" in the example above which suggests these are mutually exclusive modes?

Reply:
I find "domain_types LL DP" in real case.
 Let's use index of bit. Thanks!

Reply:
should give a bit more guidance here about which to choose for developers who may just building a standalone module are not SOF developers. LL and DP mean nothing to outsiders. To be honest I only know edf and ll, I have no clue what dp is.

Reply:
@cujomalainey Some of this is referring to ongoing work, which understandably makes this hard to follow. For DP, the PR to add this concept to SOF was just submitted for review https://github.com/thesofproject/sof/pull/7089